### ISSUE TITLE
#encryptify: End to End Encryption

### ISSUE THREAD

**Issue by robertjchen** [Expensify/App#20758]
Planning Doc: https://docs.google.com/document/d/1mLXu_8KdntiEZAxAmIt4pFl9SHFyaQ5m6IcfsUuzI94/edit#

Main tracking issue for the End to End Encryption Feature for the App. Backend changes will be tracked here: https://github.com/Expensify/Expensify/issues/269108

cc: @roryabraham

**Comment by robertjchen** [comment]
Margelo will initially help with the custom encryption library: https://github.com/Expensify/App/issues/21144

**Comment by roryabraham** [comment]
@robertjchen we have a tracking issue in E/E and E/App and also implementation issue for the front-end library. Should we consolidate this into a GitHub Project w/ a single (public) tracking issue? I'm not sure anything needs to be private for this project?

**Comment by roryabraham** [comment]
No updates for now. Let's add the planning label for now since the only place with active front-end updates right now are in https://github.com/Expensify/App/issues/21144

**Comment by roryabraham** [comment]
Go away overdue label

**Comment by roryabraham** [comment]
Chris from Margelo has started working on the Encryption lib

**Comment by robertjchen** [comment]
Chris going on vacay, requested that another team member at Margelo take over for the two weeks. Created new https://github.com/Expensify/react-native-encryptify repo.

**Comment by robertjchen** [comment]
Chris going on vacay, requested that another team member at Margelo take over for the two weeks. Created new `react-native-encryptify` repo.

**Comment by robertjchen** [comment]
No updates here. To keep things moving, I'll schedule a quick call for all of us to sync up on where things are to make sure things are on track

**Comment by robertjchen** [comment]
Chris is back but was sick, we'll sync when he is fully recovered

**Comment by chrispader** [comment]
I left an update on the encryption lib [here](https://github.com/Expensify/App/issues/21144#issuecomment-1727341757)

**Comment by roryabraham** [comment]
Nice, thanks @chrispader!

**Comment by robertjchen** [comment]
Ongoing work with encryption lib

**Comment by roryabraham** [comment]
@chrispader put up a great update for the wasm implementation of the encryption lib [here](https://github.com/Expensify/App/issues/21144#issuecomment-1754587595)

**Comment by roryabraham** [comment]
Bumped for update on wasm journey

**Comment by chrispader** [comment]
Gave an update here: https://github.com/Expensify/App/issues/21144#issuecomment-1774053158

**Comment by roryabraham** [comment]
Great updates from @chrispader last week:

- [draft PR](https://github.com/Expensify/App/pull/30146) to integrate WASM into E/App
- [web benchmarks](https://github.com/Expensify/App/pull/30523#issuecomment-1783194018) were performed there
- There are a couple draft PRs open in react-native-encryptify too:
    - https://github.com/Expensify/react-native-encryptify/pull/7
    - https://github.com/Expensify/react-native-encryptify/pull/4

**Comment by roryabraham** [comment]
@robertjchen it seems like the encryption lib is prettymuch ready-to-go, which I think means the next step might be reviving the design doc to include integration of the encryption doc. Or is there some other research still in progress?

**Comment by chrispader** [comment]
> @robertjchen it seems like the encryption lib is prettymuch ready-to-go, which I think means the next step might be reviving the design doc to include integration of the encryption doc. Or is there some other research still in progress?

Yes, everthing's ready from my side. You might want to take a look at the [example + integration PR](https://github.com/Expensify/App/pull/30146), because it already contains the necessary changes in the setup phase, to load the library (WebAssembly)

**Comment by robertjchen** [comment]
Awesome work!

**Comment by robertjchen** [comment]
Discussing next steps!

**Comment by roryabraham** [comment]
@robertjchen are we still going to try to target this by EOY or do we want to fit it into next year's roadmap?

**Comment by robertjchen** [comment]
It's going to have to be starting next year 👍

**Comment by robertjchen** [comment]
VIP project starting in the New Year!

**Comment by roryabraham** [comment]
@robertjchen Which VIP project is this being attached to?

**Comment by robertjchen** [comment]
This will be going under #vip-split

**Comment by GabiHExpensify** [comment]
@robertjchen could you clarify why/where this belongs under VIP-Split? we haven't discussed it yet in the slack channel, afaik.

**Comment by robertjchen** [comment]
@GabiHExpensify It was based on [this convo](https://expensify.slack.com/archives/C05RECHFBEW/p1700870806028159), but happy to discuss further on where it'll fit exactly.

**Comment by robertjchen** [comment]
Latest discussions on Slack [here](https://expensify.slack.com/archives/C05RECHFBEW/p1706063210530709)

**Comment by robertjchen** [comment]
Posting updated proposal from recent feedback later today and kicking off a fresh discussion.

**Comment by robertjchen** [comment]
Kicking off a fresh discussion once some other VIP/Wave priorities are out of the way (next week 🙏 )

**Comment by robertjchen** [comment]
Latest prioritization discussions on Slack: https://expensify.slack.com/archives/C05RECHFBEW/p1707971746870559

**Comment by roryabraham** [comment]
Hey @robertjchen, I'm actually going to duck out of this project for #focus. Still happy to be part of code review + conversations going forward, and don't hesitate to reach out if you need a sounding board.

**Comment by robertjchen** [comment]
Thanks!! Still holding pattern for now given existing Wave priorities

**Comment by robertjchen** [comment]
Wave items nearly out the door this week, should be able to dive back into this after that. Either late this week/next week.

**Comment by robertjchen** [comment]
I think this week is the week! Aiming to send out an updated proposal/kick off a new predesign thread before EOW.

**Comment by robertjchen** [comment]
Well, we got the wave stuff out the door by EOW 😅 Shooting for this week again, especially since I'm working on some related redacting chat functionality.

**Comment by roryabraham** [comment]
nice! Another use-case that would be good to employ E2E encryption for is whispers. Currently they're sent to everyone, and the front-end hides them from people to whom they should not be visible. we made that design in the hopes that we couple encrypt whispers with a key shared between all the intended recipients of the whisper

**Comment by robertjchen** [comment]
That makes sense! We'll incorporate that into the updated discussions now that redact functionality is out

**Comment by robertjchen** [comment]
Back from OOO and revisiting

**Comment by robertjchen** [comment]
Existing priorities taking precedence.

**Comment by robertjchen** [comment]
Some more prioritization discussions happening on Slack amidst #performance

**Comment by robertjchen** [comment]
no update

**Comment by robertjchen** [comment]
paused

**Comment by robertjchen** [comment]
on hold

**Comment by robertjchen** [comment]
on hold

**Comment by robertjchen** [comment]
on hold

**Comment by robertjchen** [comment]
on hold

**Comment by robertjchen** [comment]
on hold

**Comment by robertjchen** [comment]
not a priority at the moment

**Comment by robertjchen** [comment]
not a priority at the moment

**Comment by robertjchen** [comment]
not a priority at the moment

**Comment by robertjchen** [comment]
not a priority at the moment

**Comment by robertjchen** [comment]
No update

**Comment by robertjchen** [comment]
no update
